lib/index.js

```diff
@@ -82,6 +82,8 @@
  *   Configuration specifically for {@linkcode MarkdownHooks}.
  * @property {ReactNode | null | undefined} [fallback]
  *   Content to render while the processor processing the markdown (optional).
+ * @property {function(Root | undefined, Error | undefined): void | null | undefined} [onComplete]
```

Comment:
```suggestion
 * @property {(hast: Root | undefined, error: Error | undefined) => unknown} [onComplete]
```

Also, I think this should receive the VFile. Access to the file’s `messages` can be useful.